### DISCUSSION TITLE
DISCO root file system size (yocto)

### DISCUSSION THREAD

**Discussion by keck9939** [polarfire-soc/polarfire-soc-documentation#453] [Linux build systems for PolarFire SoC] [answered]
The pre-built 2024.02.1 Linux image for the MPFS-DISCO-KIT has essentially no free space on the root file system. As you can see from below, the partition fills the SD card, but the file system uses only a fraction of the partition.

```
root@mpfs-disco-kit:~# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
mmcblk0     179:0    0 29.8G  0 disk
|-mmcblk0p1 179:1    0   64M  0 part
|-mmcblk0p2 179:2    0    8M  0 part
`-mmcblk0p3 179:3    0 29.7G  0 part /

root@mpfs-disco-kit:~# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       659M  527M   84M  87% /
devtmpfs        273M     0  273M   0% /dev
tmpfs           274M     0  274M   0% /dev/shm
tmpfs           110M  9.9M  100M   9% /run
tmpfs           4.0M     0  4.0M   0% /sys/fs/cgroup
tmpfs           274M  8.0K  274M   1% /var/volatile

root@mpfs-disco-kit:~# fdisk -l
Disk /dev/mmcblk0: 29.81 GiB, 32010928128 bytes, 62521344 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: BFD506DD-452B-44EE-A0D4-C4B09C454B2B

Device          Start      End  Sectors  Size Type
/dev/mmcblk0p1   8192   139263   131072   64M Microsoft basic data
/dev/mmcblk0p2 139264   155647    16384    8M BIOS boot
/dev/mmcblk0p3 155648 62521303 62365656 29.7G Linux filesystem
root@mpfs-disco-kit:~#
```
I believe this could be fixed by using resize2fs, but this utility is not included in the build. If this is the case, it would be nice if the pre-built image included resize2fs. Furthermore, it would be even better if it automatically expanded on first boot e.g. like the BeagleV-Fire Linux distribution.

**Comment by vfalanis** [comment, marked as answer]
Hello @keck9939,

Could you please try adding the following line to meta-polarfire-soc-bsp/recipes-core/base-files/base-files_%.bbappend? This should resize the root partition on first boot using growfs systemd service. This will be included in our next release.

```
diff --git a/meta-polarfire-soc-bsp/recipes-core/base-files/base-files_%.bbappend b/meta-polarfire-soc-bsp/recipes-core/base-files/base-files_%.bbappend
index c91d8ea..57c2874 100644
--- a/meta-polarfire-soc-bsp/recipes-core/base-files/base-files_%.bbappend
+++ b/meta-polarfire-soc-bsp/recipes-core/base-files/base-files_%.bbappend
@@ -2,5 +2,6 @@ FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}/motd:"
 
 FILESEXTRAPATHS:prepend:icicle-kit-es := "${THISDIR}/${PN}/growfs:"
 FILESEXTRAPATHS:prepend:icicle-kit-es-amp := "${THISDIR}/${PN}/growfs:"
+FILESEXTRAPATHS:prepend:mpfs-disco-kit := "${THISDIR}/${PN}/growfs:"
 FILESEXTRAPATHS:prepend:mpfs-video-kit := "${THISDIR}/${PN}/growfs:"
```

Thanks

**Reply by keck9939** [reply]
Having used the pre-built image rather than having built it myself, I'm not presently in a position to directly try your fix, although it looks like it should solve the problem. When you release the next pre-built image, I'll try it and let you know if it doesn't work.
If I get around to building my own image before that, I'll report the result.

**Comment by Future-Electronics-Design-Center** [comment]
I tried the suggested solution and it worked well on my disco-kit.
-Tim
